src/main/java/com/hillel/items_exchange/service/impl/ImageServiceImpl.java

```diff
@@ -115,4 +121,18 @@ public void removeById(List<Long> imageIdList) {
     public void removeById(long imageId) {
         imageRepository.deleteById(imageId);
     }
+
+    private void validateImagesTypes(List<MultipartFile> images) throws UnsupportedMediaTypeException {
+        final Optional<String> unsupportedType = findUnsupportedType(images);
+        if (unsupportedType.isPresent()) {
+            throw new UnsupportedMediaTypeException("Received unsupported image type: " + unsupportedType.get());
+        }
+    }
+
+    private Optional<String> findUnsupportedType(List<MultipartFile> images) {
+        return images.parallelStream()
+                .map(MultipartFile::getContentType)
+                .filter(s -> !supportedTypes.contains(s))
+                .findAny();
+    }
 }
```

Comment:
In my opinion better to return all of given unsuppurted types or only boolean value instead of returning any one.
Also I can't test but I think it is wasteful way to make parallel stream from collection of 10 elements.

Reply:
Done